### PR TITLE
>fix `istioctl analyze` to panic when the server port in gateway is nil.

### PR DESCRIPTION
**Pull request by nicole-lihui**
**Please provide a description of this PR:**

fix bug: https://github.com/istio/istio/issues/44318

### CODE REVIEW

File: pkg/config/analysis/analyzers/gateway/conflictinggateway.go [hunk 1]
```diff
@@ -128,17 +128,17 @@ func initGatewaysMap(ctx analysis.Context) map[string]map[string][]string {
 		gw := r.Message.(*v1alpha3.Gateway)
 		gwName := r.Metadata.FullName.String()
 
-		gwSelector := klabels.SelectorFromSet(gw.Selector)
+		gwSelector := klabels.SelectorFromSet(gw.GetSelector())
 		sGWSelector := gwSelector.String()
-		for _, server := range gw.Servers {
-			sPortNumber := strconv.Itoa(int(server.Port.Number))
+		for _, server := range gw.GetServers() {
+			sPortNumber := strconv.Itoa(int(server.GetPort().GetNumber()))
```
Comment (hanxiaop) [inline pkg/config/analysis/analyzers/gateway/conflictinggateway.go:134]: Can we add a test for this?

File: releasenotes/notes/fix-44318.yaml [hunk 2]
```diff
@@ -0,0 +1,8 @@
+apiVersion: release-notes/v2
+kind: bug-fix
+area: istioctl
+issue:
+  - https://github.com/istio/istio/issues/44318
+releaseNotes:
+  - |-
+    **Fixed** `istioctl analyze` to panic when the server port in gateway is nil.
```
Comment (hanxiaop) [inline releasenotes/notes/fix-44318.yaml:8]:
```suggestion
    **Fixed** `istioctl analyze` to prevent panic when the server port in Gateway is nil.
```

**Review by hzxuzhonghu — COMMENTED** [review]
I think we should also add validation for server.port nil

File: releasenotes/notes/fix-44318.yaml (hunk 2 above)
Comment (ymesika) [reply releasenotes/notes/fix-44318.yaml:8]:
```suggestion
    **Fixed** a panic in `istioctl analyze` when the server port in gateway is not set.
```

Comment (hanxiaop) [reply releasenotes/notes/fix-44318.yaml]: I think you can apply @ymesika 's suggestion since it's more concise

**Comment by nicole-lihui** [comment]
> I think we are missing check for server.port nil in validation.
@hzxuzhonghu 

gateway.servers[*].port nil check already exists, ref: 
https://github.com/istio/istio/blob/704afa70f64c6efc886793453ea70b07e9a89d27/pkg/config/validation/validation.go#L467-L522
https://github.com/istio/istio/blob/704afa70f64c6efc886793453ea70b07e9a89d27/pkg/config/validation/validation.go#L549-L564

**Review by hzxuzhonghu — APPROVED** [review]

**Comment by hanxiaop** [comment]
@nicole-lihui It seems like tests are missing, can you also add test cases to cover this?

**Pull request merged**